src/components/gltf-model-plus.js

```diff
@@ -194,33 +202,122 @@ function getFilesFromSketchfabZip(src) {
   });
 }
 
-function cachedLoadGLTF(src, basePath, contentType, preferredTechnique, onProgress) {
-  if (!GLTFCache[src]) {
-    GLTFCache[src] = new Promise(async (resolve, reject) => {
-      try {
-        let gltfUrl = src;
-        let onLoad = resolve;
-        if (contentType === "model/gltf+zip") {
-          const fileMap = await getFilesFromSketchfabZip(src);
-          gltfUrl = fileMap["scene.gtlf"];
-          onLoad = model => {
-            // The GLTF is now cached as a THREE object, we can get rid of the original blobs
-            Object.keys(fileMap).forEach(URL.revokeObjectURL);
-            resolve(model);
-          };
-        }
+async function loadEnvMap() {
+  const urls = [cubeMapPosX, cubeMapNegX, cubeMapPosY, cubeMapNegY, cubeMapPosZ, cubeMapNegZ];
+  const texture = await new THREE.CubeTextureLoader().load(urls);
+  texture.format = THREE.RGBFormat;
+  return texture;
+}
+
+function resolveURL(url, path) {
+  // Invalid URL
+  if (typeof url !== "string" || url === "") return "";
+
+  // Absolute URL http://,https://,//
+  if (/^(https?:)?\/\//i.test(url)) return url;
+
+  // Data URI
+  if (/^data:.*,.*$/i.test(url)) return url;
+
+  // Blob URL
+  if (/^blob:.*$/i.test(url)) return url;
+
+  // Relative URL
+  return path + url;
+}
+
+async function loadGLTF(src, preferredTechnique, onProgress) {
+  const { raw, origin, contentType } = await resolveMedia(src);
+  const basePath = THREE.LoaderUtils.extractUrlBase(origin);
+
+  let gltfUrl = raw;
+  let fileMap;
+
+  if (contentType === "model/gltf+zip") {
+    fileMap = await getFilesFromSketchfabZip(gltfUrl);
+    gltfUrl = fileMap["scene.gtlf"];
+  }
+
+  const gltfLoader = new THREE.GLTFLoader();
+  gltfLoader.setPath(basePath);
+  gltfLoader.setLazy(true);
+
+  const { parser } = await new Promise((resolve, reject) => gltfLoader.load(gltfUrl, resolve, onProgress, reject));
+
+  const json = parser.json;
+  const images = json.images;
+  const buffers = json.buffers;
+  const materials = json.materials;
+
+  const pendingFarsparkPromises = [];
+
+  if (images) {
+    for (const image of images) {
```

Comment:
I would probably use map here. Even if it does create garbage this isn't really in a night loop so I would guess it would be fine, haven't ever profiled it to see how bad it is though.

Reply:
I see no benefit to using map here.

Reply:
It is looping through an array and mapping it into another array. It just cleans up some of the ceremony. Not a big deal either way.